.github/workflows/deploy.yml

```
@@ -27,17 +30,46 @@ jobs:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          description: "Deploying ezinn-client to staging via Docker"
+  deploy:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Inject environment variables
+        run: echo "${{ secrets.ENV_FILE }}" > .env
+
+      - name: Determine environment
+        id: env
+        run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:2:28: Double quote to prevent globbing and word splitting

(shellcheck)